File: server/admin/schemas/question.py
from typing import Optional
from pydantic import BaseModel
from common.schemas import SearchSchema


class QuestionCreateSchema(BaseModel):
    # 问题类型
    # 数学：单选、填空、解答、判断题、步骤题、图形题
    # 英语：选择、完型、阅读理解、写作、语法填空
    type: str
    # 问题内容
    content: str
    # 选项
    options: Optional[list[str]] = None
    # 答案
    answer: str
    # 年级（1 ～ 12）
    grade: int
    # 科目
    subject: str
    # 知识点 ID
    knowledge_id: Optional[int] = None
    # 课程单元 ID
    course_unit_id: Optional[int] = None
    # 教材 ID
    textbook_id: Optional[int] = None
    # 问题来源：AI生成 | 后台创建 ｜ 用户上传
    source: str


class QuestionUpdateSchema(BaseModel):
    type: Optional[str] = None
    content: Optional[str] = None
    options: Optional[list[str]] = None
    answer: Optional[str] = None
    grade: Optional[int] = None
    subject: Optional[str] = None
    knowledge_id: Optional[int] = None
    course_unit_id: Optional[int] = None
    textbook_id: Optional[int] = None
    source: Optional[str] = None
    status: Optional[int] = None


class QuestionSearchSchema(SearchSchema):
    type: Optional[str] = None
    grade: Optional[int] = None
    subject: Optional[str] = None
    knowledge_id: Optional[int] = None
    course_unit_id: Optional[int] = None
    source: Optional[str] = None
    status: Optional[int] = None

File: packages/shared-web/src/types/question.ts
/**
 * 题目相关类型定义
 */

import type { SearchParams } from './common';
import type { Textbook, Unit } from './textbook';

/**
 * 题目信息（对应 QuestionSchema）
 */
export interface Question {
  id: string | number;
  type: string; // 题目类型（主类型）
  subtype?: string; // 题目子类型
  subject: string; // 科目
  grade: number; // 年级
  content: string; // 题目内容
  options?: string; // 选项（多行文本）
  answer?: string; // 答案
  resource?: string; // 资源路径（图片/音频URL）
  difficulty?: string; // 难度
  resource_type?: 'image' | 'audio' | string | null; // 资源类型
  resource_content?: string; // 资源内容（录音文本等）
  textbook_id?: number;
  unit_id?: number; // 单元ID
  knowledge?: string; // 知识点
  // 前端扩展字段
  is_correct?: boolean; // 是否答对（答题后）
  order?: number; // 题目顺序（练习会话中）
  textbook?: Textbook;
  unit?: Unit;
}

/**
 * 题目创建表单（对应 CreateQuestionSchema）
 */
export interface QuestionCreateSchema {
  subject: string;
  grade: number;
  type: string;
  content: string;
  options?: string;
  answer?: string;
  resource?: string;
  difficulty?: string;
  knowledge_id?: number;
  unit_id?: number;
  textbook_id?: number;
}

/**
 * 题目更新表单（对应 UpdateQuestionSchema）
 */
export interface QuestionUpdateForm {
  subject?: string;
  grade?: number;
  type?: string;
  subtype?: string;
  content?: string;
  options?: string | string[];
  answer?: string;
  resource?: string;
  resource_type?: string;
  resource_content?: string;
  difficulty?: string;
  knowledge?: string;
  unit_id?: number;
  textbook_id?: number;
}

/**
 * 题目搜索参数（对应 SearchQuestionSchema）
 */
export interface QuestionSearchParams extends SearchParams {
  question_id?: number;
  keyword?: string;
  textbook_id?: number;
  unit_id?: number;
  type?: string;
  subtype?: string;
  difficulty?: string;
  subject?: string;
  grade?: number;
  resource_type?: string;
  resource_generated?: boolean;
}

